borsh/src/schema/container_ext/validate.rs

```diff
@@ -78,9 +92,7 @@ fn validate_impl<'a>(
             tag_width,
             variants,
         } => {
-            if *tag_width > 8 {
-                return Err(Error::TagTooWide(declaration.to_string()));
-            }
+            check_tag_width(declaration, *tag_width, variants.len() as u64)?;
```

Comment:
using `variants.len()` is incorrect here, as it should be `variants.find_max_discriminant()` of `i64` type, and info about discriminants is not collected into `variants`